test/test-lib-nodejs.sh

```diff
@@ -528,19 +528,59 @@ function test_safe_logging() {
   fi
 }
 
+
+function ct_pull_or_import_postgresql() {
+  postgresql_image="quay.io/sclorg/postgresql-12-c8s"
+  image_short="postgresql:12"
+  image_tag="${image_short}"
+  # Variable CVP is set by CVP pipeline
+  if [ "${CVP:-0}" -eq "0" ]; then
+    # In case of container or OpenShift 4 tests
+    # Pull image before going through tests
+    # Exit in case of failure, because postgresql container is mandatory
+    ct_pull_image "${postgresql_image}" "true"
+  else
+    # Import postgresql-10-centos7 image before running tests on CVP
+    oc import-image "${image_short}:latest" --from="${postgresql_image}:latest" --insecure=true --confirm
+    # Tag postgresql image to "postgresql:10" which is expected by test suite
+    oc tag "${image_short}:latest" "${image_tag}"
+  fi
+}
+
 # Check the imagestream
-function test_nodejs_imagestream() {
+function test_nodejs_imagestream_with_no_db() {
+  local ret_val=0
   if [[ "${VERSION}" == *"minimal"* ]]; then
     VERSION=$(echo "${VERSION}" | cut -d "-" -f 1)
   fi
   ct_os_test_image_stream_quickstart \
     "${THISDIR}/imagestreams/nodejs-${OS%[0-9]*}.json" \
-    "https://raw.githubusercontent.com/sclorg/nodejs-ex/master/openshift/templates/nodejs.json" \
+    "https://raw.githubusercontent.com/sclorg/nodejs-ex/${BRANCH_TO_TEST}/openshift/templates/nodejs.json" \
     "${IMAGE_NAME}" \
     'nodejs' \
-    "Welcome to your Node.js application on OpenShift" \
+    "Node.js Crud Application" \
     8080 http 200 \
-    "-p SOURCE_REPOSITORY_REF=master -p SOURCE_REPOSITORY_URL=https://github.com/sclorg/nodejs-ex.git -p NODEJS_VERSION=${VERSION} -p NAME=nodejs-testing"
+    "-p SOURCE_REPOSITORY_REF=${BRANCH_TO_TEST} -p SOURCE_REPOSITORY_URL=https://github.com/sclorg/nodejs-ex.git -p NODEJS_VERSION=${VERSION} -p NAME=nodejs-testing" || ret_val=1
+    return $ret_val
+}
+
+function test_nodejs_imagestream_with_db() {
```

Comment:
I don't understand what this test is supposed to verify -- I thought image streams are the same no matter whether the example app is run with a DB or without. I must be missing something, so maybe a comment why we need this case would be useful.

Reply:
@hhorak Basically we have two approaches as far as I remember.
* First of all, we have function called `template` which means, that we do not use imagestreams, but only images, that are imported to the openshift by a specific tag
* The second approach is a function called `imagestream`, like this one,  that imports all imagestreams, and the correct one is used in the corresponding template.
I am hoping that I provided a full description.

Reply:
I think Honza had in mind that in this PR **two tests for templates** and **two tests for imagestreams** are being introduced. I am also missing why both (with_db&with_no_db) are needed.
I think you explained that imagestreams and templates should be tested, and I agree with that. Could you please also explain us why the tests are doubled?

Reply:
What I'm missing is whether there is a need to have both `test_nodejs_imagestream_with_no_db` and 
`test_nodejs_imagestream_with_db` -- when we need to test an imagestream, it should be reasonably well tested with just one of those.

Reply:
Because there are parameters for database, but I can add there as a string.
In case we add parameters to database to template where parameters are not expected, then the test will fail.

Reply:
I see your point. E.g. in https://github.com/sclorg/s2i-python-container/blob/master/test/test-lib-python.sh we have only with database. Let's get remove without DB.

Reply:
Removed and rebased by https://github.com/sclorg/s2i-nodejs-container/pull/419/commits/acd08c71628528a19f25f0d84bee14a73856e83b